kumascript/src/api/page.ts

```diff
@@ -38,8 +40,8 @@ const page = {
   //
   // This is not called by any macros, and is only used here by
   // wiki.tree(), so we could move it to be part of that function.
-  subpages(path, depth, self) {
-    return this.page.subpagesExpand(path, depth, self);
+  subpages(this: KumaThis, path, depth, self) {
+    return (this.page as any).subpagesExpand(path, depth, self);
```

Comment:
I don't know why `this.page` doesn't work, possibly because `this.page` refers to what `this` would be without specifying `this: KumaThis`. 🤷

Reply:
Haven't tried, but it looks like all those methods get called using [`Function.call()`](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Function/call) with the environment ("KumaThis") as `argThis`.

Reply:
`this.page.XXX` is needed because `this` is not the `page` when this function is called, as per the above statement.  TypeScript doesn't like this since we're essentially overwriting the variable with a new value, hence the need to cast to `any` here.